File: testdata/bench/kotlin/BenchmarkTest00120.kt
package bench

import io.ktor.server.application.*
import io.ktor.server.response.*
import java.net.URI

// Safe: strip protocol and validate host before redirect
suspend fun handler00120(call: ApplicationCall) {
    val url = call.request.queryParameters["goto"] ?: "/"
    val parsed = try { URI(url) } catch (e: Exception) { URI("/") }
    val safe = if (parsed.host == null || parsed.host == "example.com") url else "/"
    call.respondRedirect(safe)
}

File: testdata/bench/go/BenchmarkTest00120.go
package bench

import (
	"fmt"
	"net/http"
	"net/url"
)

// url.QueryEscape for attribute
func Handler00120(w http.ResponseWriter, r *http.Request) {
	q := r.URL.Query().Get("q")
	safe := url.QueryEscape(q)
	w.Header().Set("Content-Type", "text/html")
	fmt.Fprintf(w, "<a href='/search?q=%s'>Search</a>", safe)
}
